### PR TITLE
Deal dmg to target via the damage card(s)

### PR DESCRIPTION
**Pull request by thul**
This MR adds automated dmg dealing to the first target of a player. Using the Weapon roll, damage roll or psychic roll. Target resilience and armor rating is taken into account. If dmg is more than resilience target takes damage, if it's equal they take 1 shock dmg and if it's lower they don't take any damage.

Only the first target will be damaged, multi-target attacks are not supported (yet).

### CODE REVIEW

**Comment by moo-man** [comment]
Thanks for the PR. However generally I prefer to follow the common workflow of damage application in systems by right clicking the chat card to apply damage to selected (not targeted) tokens.

**Comment by thul** [comment]
I didn't know that was common practice. I'll close this PR then

**Comment by thul** [comment]
This PR now adds 3 options

1. Apply Damage - Default, AP will remove resilience too
2. Apply Damage (Invulnerable armour) - When target(s) is/are wearing invuln armor that can't have AP removed
3. Apply Damage (AP only on Armour) - AP will only remove Armour rating and not resilience itself

Right clicking the damage card when 1 or more tokens are targeted will make the options appear

File: template/chat/damage.html [hunk 1]
```diff
@@ -10,7 +10,7 @@ <h3>{{name}}</h3>
     </div>
 
     <div class="wrapper-input">
-        <p><strong>{{localize "WEAPON.DAMAGE"}}: {{result.damage.total}}</strong></p>
+        <p><strong>{{localize "WEAPON.DAMAGE"}}: {{result.damage.total}}</span></strong></p>
```
Comment (moo-man) [inline template/chat/damage.html:13]: I don't think this `</span>` has a opening element

Comment (thul) [reply template/chat/damage.html]: Woops sorry, that was an artifact from older attempts. removed it now

File: scripts/hooks/entryContext.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+function _computeArmour(actor) {
+    let foundItems = actor.data.items.filter(a => a.data.type === "armour");
+    let armourRating = 0;
+    for (let armour of foundItems) {
+        armourRating += armour.rating;
+    }
+    return armourRating;
+}
```
Comment (moo-man) [inline scripts/hooks/entryContext.js:84]: Does this need to check for anything? I don't see anything in the template about whether armour is actively being worn or not. Is this a limitation of the current system?

Comment (thul) [reply scripts/hooks/entryContext.js:84]: There is no flag that indicates gear is currently (not) worn. So i assume anything that is on the character sheet and is of the "Armour" type is being worn.

File: scripts/hooks/entryContext.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    game.user.targets.forEach(t => _dealDamageToTarget(rollData, t.actor, false));
+                }
+            },
+            {
+                name: "Apply Damage (AP only on Armour)",
+                icon: '<i class="fas fa-user-times"></i>',
+                condition: canApply,
+                callback: li => {
+                    let rollData = game.messages.get(li.attr("data-message-id")).data.flags.rolldata;
+                    game.user.targets.forEach(t => _dealDamageToTarget(rollData, t.actor, true, true));
```
Comment (moo-man) [inline scripts/hooks/entryContext.js:29]: Could these names be localized?

Comment (thul) [reply scripts/hooks/entryContext.js:29]: Can do

**Pull request merged**